### PR TITLE
WFLY-13809 Get application name from BatchEnvironment instead of jndi lookup and Upgrade jberet-core from 1.3.9.Final to 1.3.10.Final; from 2.0.0.Final to 2.0.1.Final in ee-9 preview

### PR DESCRIPTION
**Pull request by parsharma**
Issue: https://issues.redhat.com/browse/WFLY-13809

Related Issue: https://issues.redhat.com/browse/WFLY-15059 
https://issues.redhat.com/browse/JBERET-495

This PR also contains a component upgrade.

### CODE REVIEW

**Comment by chengfang** [comment]
@parsharma can you add to the title, to make it clear this also contains a component upgrade?

    WFLY-15059

Upgrade jberet-core from 1.3.9.Final to 1.3.10.Final; from 2.0.0.Final to 2.0.1.Final in ee-9 preview

File: batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/DefaultBatchEnvironment.java [hunk 1]
```diff
@@ -76,4 +76,9 @@ public JobXmlResolver getJobXmlResolver() {
     public Properties getBatchConfigurationProperties() {
         throw BatchLogger.LOGGER.noBatchEnvironmentFound(WildFlySecurityManager.getCurrentContextClassLoaderPrivileged());
     }
+
+    @Override
+    public String getApplicationName() {
+        return null;
+    }
```
Comment (chengfang) [inline batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/DefaultBatchEnvironment.java:83]: This method should also throw the same exception as other methods.

File: pom.xml [hunk 2]
```diff
@@ -394,7 +394,7 @@
         <version.org.infinispan.protostream>4.4.1.Final</version.org.infinispan.protostream>
         <version.org.jasypt>1.9.3</version.org.jasypt>
         <version.org.javassist>3.23.2-GA</version.org.javassist>
-        <version.org.jberet>1.3.9.Final</version.org.jberet>
+        <version.org.jberet>1.3.10.Final</version.org.jberet>
```
Comment (chengfang) [inline pom.xml:397]: ee-9/feature-pack/pom.xml should also be changed to contain an upgrade of jberet-core from 2.0.0.Final to 2.0.1.Final.

File: batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/DefaultBatchEnvironment.java (hunk 1 above)
Comment (parsharma) [reply batch-jberet/src/main/java/org/wildfly/extension/batch/jberet/DefaultBatchEnvironment.java:83]: Changes done.

**Review by chengfang — APPROVED** [review]

File: ee-9/feature-pack/pom.xml [hunk 3]
```diff
@@ -54,6 +54,7 @@
         <version.jakarta.authentication.jakarta-authentication-api>2.0.0</version.jakarta.authentication.jakarta-authentication-api>
         <version.jakarta.authorization.jakarta-authorization-api>2.0.0</version.jakarta.authorization.jakarta-authorization-api>
         <version.jakarta.batch.jakarta.batch-api>2.0.0</version.jakarta.batch.jakarta.batch-api>
+        <version.org.jberet>2.0.1.Final</version.org.jberet>
```
Comment (bstansberry) [inline ee-9/feature-pack/pom.xml:57]: @parsharma This property is already declared in line 93 of this file. Please update that one.

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
WildFly Preview tests:

https://ci.wildfly.org/viewLog.html?buildId=267207&tab=buildResultsDiv&buildTypeId=WF_Ee9BootableJarLinuxJdk8
https://ci.wildfly.org/viewLog.html?buildId=267209&tab=buildResultsDiv&buildTypeId=WF_WildFlyPreviewLinuxJdk11

**Pull request merged**